VirtualH89/Src/z80.cpp

```diff
@@ -1533,37 +1538,42 @@ Z80::addClockTicks()
 }
 
 void
-Z80::setSpeed(bool fast)
-{
-    debugss(ssZ80, WARNING, "%s: - %d\n", __FUNCTION__, fast);
+Z80::setSpeedup(int factor) {
+    speedUpFactor_m = factor;
+}
 
-    if (fast)
-    {
+void
+Z80::enableFast() {
+    GppListener::addListener(this);
+}
+
+void
+Z80::gppNewValue(BYTE gpo) {
+    fast_m = ((gpo & z80_gppSpeedSelBit_c) != 0);
```

Comment:
When I saw you added the fast_m, I thought you would use that to verify whether or not a speed change was needed. I.e., if it was already set to fast, and the passed in gpp value is fast, then do nothing.